compiler/rustc_resolve/src/access_levels.rs

```diff
@@ -75,30 +71,20 @@ impl<'r, 'a> AccessLevelsVisitor<'r, 'a> {
         let module_level = self.r.access_levels.map.get(&module_id).copied();
         assert!(module_level >= Some(AccessLevel::Exported));
 
-        if let Some(exports) = self.r.reexport_map.get(&module_id) {
-            let pub_exports = exports
-                .iter()
-                .filter(|ex| ex.vis == Visibility::Public)
-                .cloned()
-                .collect::<Vec<_>>();
-
-            let module = self.r.get_module(module_id.to_def_id()).unwrap();
-            for export in pub_exports.into_iter() {
-                if let Some(export_def_id) = export.res.opt_def_id().and_then(|id| id.as_local()) {
-                    self.set_access_level_def_id(export_def_id, Some(AccessLevel::Exported));
-                }
+        let module = self.r.get_module(module_id.to_def_id()).unwrap();
+        let resolutions = self.r.resolutions(module).clone();
```

Comment:
Cloning the resolution table is an expensive operation, you have to borrow it in some way.